csv2wiki/__init__.py

```diff
@@ -603,31 +615,33 @@ def __init__(self, config, csv_input, null_as_value, msg_out, dry_run_out):
             # There are two kinds of lines:
             #
             #   1) New section indicator (starts with dots)
-            #   2) Column number (starts with a number)
+            #   2) Text line (starts with a pipe)
+            #   3) Comment line (starts with a pound sign)
+            #      - This is not handled explicitly here because
+            #        it's handled by the python config parser
+            #        before we get the sec_map
             #
             # These regexps help us figure out which kind we've got.
             dot_matcher = re.compile("^(\\.+)\\s*(.*)$")
-            col_matcher = re.compile("^(([0-9]+\\s*)+)\\s*.*$")
+            txt_matcher = re.compile("^\\|\\s*(.*)$")
 
             # Because of the way Python parses ConfigParser syntax,
             # the format we get the sec_map in is one big string,
             # splittable on line breaks into a list of lines.
             for line in sec_map.splitlines():
                 # As usual, I wish Python had Lisp-style 'cond'.
-                m = dot_matcher.match(line)
-                if m:
+                if dot_matcher.search(line):
+                    m = dot_matcher.match(line)
                     self._section_structure.append(
                         WikiSectionSkel(m.group(1).count("."),
                                         m.group(2) or ""))
+                elif txt_matcher.search(line):
```

Comment:
Don't hit me again on the same bruise -- ouch! :-)

Reply:
POW